task/coverity-availability-check/0.1/coverity-availability-check.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: sast-coverity-check
+spec:
+  description: >-
+    This task performs needed checks in order to use Coverity image in the pipeline. It will check for a Coverity license secret and an authentication secret for pulling the image.
+  results:
+    - description: Tekton task result output.
+      name: TASK_OUTPUT
+  params:
+    - name: COV_LICENSE
+      description: Name of secret which contains the Coverity license
+      default: cov-license
+    - name: AUTH_TOKEN_COVERITY_IMAGE
+      description: Name of secret which contains the authentication token for pulling the Coverity image.
+      default: "auth-token-coverity-image"
+  volumes:
+    - name: cov-license
+      secret:
+        secretName: $(params.COV_LICENSE)
+        optional: true
+    - name: auth-token-coverity-image
+      secret:
+        secretName: $(params.AUTH_TOKEN_COVERITY_IMAGE)
+        optional: true
+  steps:
+    - name: coverity-availability-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.7@sha256:cf6808a3bd605630a5d9f20595ff7c43f8645c00381219d32f5a11e88fe37072
+      onError: stopAndFail
```

Comment:
I'm not sure about this,
we want tasks to pass in all cases, failures should be reported using results.

Reply:
We use the `coverity-availability-check` only to decide whether the `sast-coverity-check` (the actual scanning task) can ever be run or not.  If I remember correctly, this is a solution suggested by @14rcole.

Reply:
Agree with @jsztuka we should make tekton task succeeds always, refer to [ADR 0014-let-pipelines-proceed](https://github.com/konflux-ci/architecture/blob/main/ADR/0014-let-pipelines-proceed.md), but can have `ERROR` or `FAIL` in tekton test result like [this task](https://github.com/konflux-ci/build-definitions/blob/main/task/inspect-image/0.1/inspect-image.yaml#L80-L82).  This tekton result will be consumed by later EC pipeline. And at the same time your later `sast-coverity-check` can have some check steps before running real coverity check.

Reply:
> And at the same time your later sast-coverity-check can have some check steps before running real coverity check.

@hongweiliu17 This is actually not true.  `sast-coverity-check` uses a container image that cannot be pulled without an authentication token.  If the `sast-coverity-check` is triggered and the authentication token is not available, it causes the whole pipeline to fail.  This is actually the only reason why the `coverity-availability-check` task ever exists.

Reply:
So the `when` in pipelinerun or other check in later task can resolve it. Take one example, the files generated by [inspect-image](https://github.com/konflux-ci/build-definitions/blob/main/task/inspect-image/0.1/inspect-image.yaml) task is depended by later tasks, but we still make inspect-image task completed always even its script meet some error.

Reply:
@hongweiliu17 I am not the author of this pull request (@jperezdealgaba is on PTO right now).  Nevertheless, this is a solution that was designed in cooperation with the Konflux team (represented by @14rcole back then).  The first two suggestions did not work as expected, which is the reason why we ended up with the separate `coverity-availability-check` task.

Reply:
I think `when`  is not a good way to do it, when will cause the task skipped. the pipelinerun result will not be the expected one.
I still feel the reason of failure should be emitted to tekton result `TEST_OUTPUT` like other tasks. Refer to this [integration-service-api-contracts](https://github.com/konflux-ci/architecture/blob/main/ADR/0013-integration-service-api-contracts.md#tekton-task-results)

Reply:
@jperezdealgaba The task should exit but should not fail.  If the task fails then the whole build pipeline will terminate early, which will prevent other checks from being run